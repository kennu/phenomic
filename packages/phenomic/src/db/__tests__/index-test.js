const db = require("..")

describe("db", () => {

  beforeEach(() => {
    jest.resetModuleRegistry()
  })

  it("should be able to put & get a value", () => {
    return db.put("foo", "bar", { data: { title: "bar" } })
      .then(() => db.get("foo", "bar"))
      .then(value => expect(value).toMatchSnapshot())
  })

  it("should throw when value isn't there", () => {
    return db.get("foo", "baz")
      .catch(error => expect(error).toMatchSnapshot())
  })

})
