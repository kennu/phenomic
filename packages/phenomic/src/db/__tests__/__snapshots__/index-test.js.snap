exports[`db should be able to put & get a value 1`] = `
Object {
  "key": "bar",
  "value": Object {
    "body": undefined,
    "title": "bar",
  },
}
`;

exports[`db should throw when value isn't there 1`] = `[NotFoundError: Key not found in database]`;
